test(tasks/coverage/typescript): Update `2541-2999` error codes handling

sapphi-red opened the pull request:
Part of https://github.com/oxc-project/oxc/issues/11582

Used the same hash in https://github.com/oxc-project/oxc/issues/11582#issuecomment-3008344585 (261630d650c0c961860187bebc86e25c3707c05d).

Hope you don't mind 😀

sapphi-red commented:
* **#15496** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15496?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#15495** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15495?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#15494** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15494?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#15493** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15493?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#15492** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15492?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15492?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=15492).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

sapphi-red commented on tasks/coverage/src/typescript/constants.rs:
```diff
@@ -206,7 +206,195 @@ pub static NOT_SUPPORTED_ERROR_CODES: phf::Set<&'static str> = phf::phf_set![
+    "2550", // Property 'setBigInt64' does not exist on type 'DataView<ArrayBuffer>'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2020' or later.
+    "2551", // Property 'fng2' does not exist on type 'typeof A'. Did you mean 'fng'?
+    "2552", // Cannot find name 'tupel'. Did you mean 'tuple'?
+    "2554", // Expected 0 arguments, but got 1.
```
Cases like
```
foo("", 0);
    ~~~
!!! error TS2554: Expected 3 arguments, but got 2.
```

sapphi-red commented on tasks/coverage/src/typescript/constants.rs:
```diff
@@ -206,7 +206,195 @@ pub static NOT_SUPPORTED_ERROR_CODES: phf::Set<&'static str> = phf::phf_set![
+    "2559", // Type 'D' has no properties in common with type 'C'.
+    "2560", // Value of type '() => { timeout: number; }' has no properties in common with type 'Settings'. Did you mean to call it?
+    "2561", // Object literal may only specify known properties, but 'colour' does not exist in type 'CSSProps'. Did you mean to write 'color'?
+    "2562", // Base class expressions cannot reference class type parameters.
```
```
    class Gen<T> extends base<T>() {}  // Error, T not in scope
                              ~
!!! error TS2562: Base class expressions cannot reference class type parameters.
```
On second though, should this be not included in this list?

leaysgur replied:
When we're unsure, just leave it as it is. 🙈

sapphi-red commented on tasks/coverage/src/typescript/constants.rs:
```diff
@@ -206,7 +206,195 @@ pub static NOT_SUPPORTED_ERROR_CODES: phf::Set<&'static str> = phf::phf_set![
+    "2628", // Cannot assign to 'A' because it is an enum.
+    "2629", // Cannot assign to 'f' because it is a class.
+    "2630", // Cannot assign to 'foo' because it is a function.
+    "2631", // Cannot assign to 'M' because it is a namespace.
```
Same with `enum` above.

sapphi-red commented on tasks/coverage/src/typescript/constants.rs:
```diff
@@ -206,7 +206,195 @@ pub static NOT_SUPPORTED_ERROR_CODES: phf::Set<&'static str> = phf::phf_set![
+    "2686", // 'Puppeteer' refers to a UMD global, but the current file is a module. Consider adding an import instead.
+    "2687", // All declarations of 'x' must have identical modifiers.
+    "2688", // Cannot find type definition file for 'react'.
+    "2689", // Cannot extend an interface 'Base'. Did you mean 'implements'?
```
Cases like
```
    interface Base<T, U> {
        x: T;
        y: U;
    }
    
    // Error, no Base constructor function
    class D0 extends Base<string, string> {
                     ~~~~
!!! error TS2689: Cannot extend an interface 'Base'. Did you mean 'implements'?
    }
```
This requires to know whether `Base` is a type or a value.

sapphi-red commented on tasks/coverage/src/typescript/constants.rs:
```diff
@@ -206,7 +206,195 @@ pub static NOT_SUPPORTED_ERROR_CODES: phf::Set<&'static str> = phf::phf_set![
+    "2876", // This relative import path is unsafe to rewrite because it looks like a file name, but actually resolves to "./foo.ts/index.ts".
+    "2877", // This import uses a '.ts' extension to resolve to an input TypeScript file, but will not be rewritten during emit because it is not a relative path.
+    "2879", // Using JSX fragments requires fragment factory 'React' to be in scope, but it could not be found.
+    "2880", // Import assertions have been replaced by import attributes. Use 'with' instead of 'assert'.
```
Should this be excluded from this list?

sapphi-red commented:
I've removed the two from the list and updated the snapshot 👍

sapphi-red committed:
test(tasks/coverage/typescript): Update `2541-2999` error codes handling (#15492)

Part of https://github.com/oxc-project/oxc/issues/11582

Used the same hash in https://github.com/oxc-project/oxc/issues/11582#issuecomment-3008344585 (261630d650c0c961860187bebc86e25c3707c05d).

Hope you don't mind 😀
